### PR TITLE
Upgrade Node.js Localize module deps and align with current API [Dev-1682]

### PR DESCRIPTION
**Pull request by Binnsler**
This PR makes a number of updates:

1. Upgraded Node modules to match what we use in the Localize project. Note that I did not yet upgrade from the `request` library to `axios` because despite it's deprecation `request` is download weekly about 5-10 million times more than `axios`.

2. Upgraded the module to match the current API

3. Fixed the test suite so it will run properly once given an API key and project key

4. Updated the changelog

5. Updated the README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,16 +20,17 @@ Endpoints
...
 **localizeServices.project**
 
     localizeService.project.create({
             name: 'project1',
             sourceLanguage: 'en',
+            type: 'web'
         }, callback);
 
-    Required fields:  name, sourceLanguage       
+    Required fields:  name, sourceLanguage, type
```
Comment (shinyjoseph) [inline README.md:33]: Type of project need not be mandatory. It defaults `web`. The document should also be updated. It marks type as mandatory, but the explanation reads `defaults to web if not specified.`  https://help.localizejs.com/reference/api-projects-create

Comment (shinyjoseph) [reply README.md:33]: Also, should we make `activeLanguages` mandatory in the backend API? It is mandatory while creating projects through dashboard https://github.com/Localize/Localize/blob/master/services/dashboard/controllers/project.js#L44. But, not mandatory while it is created through rest API https://github.com/Localize/Localize/blob/master/services/api/controllers/project.js#L45-L49.

File: main.js [hunk 2]
```diff
@@ -269,8 +270,9 @@ module.exports = function (apiKey) {
     content: {
       // Import contents
       import: (data, done) => {
-        if (!data.projectKey || !data.language || !data.format || !data.type || !data.file) return done(new Error('Invalid input params'));
+        if (!data.projectKey || !data.language || !data.format || !data.type || !data.content) return done(new Error('Invalid input params'));
         const endPoint = 'projects/' + data.projectKey + '/resources';
+        console.log('data', data);
```
Comment (shinyjoseph) [inline main.js:275]: console to be removed

**Review by shinyjoseph — COMMENTED** [review]
@Binnsler  I tested the npm module and it is working fine. 
I just have a couple of questions/clarification posted. Kindly let me know your thoughts on it. Thanks!

cc:/ @paton

File: README.md (hunk 1 above)
Comment (Binnsler) [reply README.md]: I fixed your first comment and created a task for the second comment: https://shapesuite.atlassian.net/browse/PBI-1026

File: main.js (hunk 2 above)
Comment (Binnsler) [reply main.js]: Removed

**Pull request merged**